### PR TITLE
Jacky update django spring24

### PR DESCRIPTION
**Pull request by JackyJerkyYt**
update the website to spring 2024, as said on Monday, we don't need to use the final dates as we don't use them, so I didn't include them

### CODE REVIEW

File: backend/berkeleytime/config/semesters/spring2024.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# SIS API Keys
+SIS_TERM_ID = 2242
+
+TELEBEARS = {
+    'phase1_start': datetime.datetime(2023, 10, 24),
+    'phase1_end': datetime.datetime(2023, 11, 5),
+    'phase2_start': datetime.datetime(2023, 11, 22),
+    'phase2_end': datetime.datetime(2023, 1, 7),
+    'adj_start': datetime.datetime(2023, 1, 8),
```
Comment (maxmwang) [inline backend/berkeleytime/config/semesters/spring2024.py:18]: Should be 2024.

**Review by maxmwang — CHANGES_REQUESTED** [review]

File: backend/berkeleytime/config/semesters/spring2024.py (hunk 1 above)
Comment (Youngster-Joey) [reply backend/berkeleytime/config/semesters/spring2024.py]: also i think phase1 and phase2 start should be 10/16 and 11/14 respectively

File: frontend/src/utils/scheduler/scheduler.ts [hunk 2]
```diff
@@ -175,8 +175,8 @@ export const serializeSchedule = (schedule: Schedule, semester: Semester): Backe
  * Generates iCal file string.
  */
 export function scheduleToICal(schedule: Schedule, semester: Semester): string {
-	const SEMESTER_START = new Date(2023, 7, 23);
-	const LAST_COURSE_DAY = new Date(2023, 11, 1);
+	const SEMESTER_START = new Date(2024, 1, 16);
+	const LAST_COURSE_DAY = new Date(2024, 4, 26);
```
Comment (kevinzwang) [inline frontend/src/utils/scheduler/scheduler.ts:179]: Javascript `Date` month values go from 0 to 11 (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date)

**Review by kevinzwang — CHANGES_REQUESTED** [review]
Please include a finals config file as well. It may not be used in the frontend but it is depended on in [other parts of the code](https://github.com/asuc-octo/berkeleytime/blob/master/backend/catalog/mapper/section.py#L103). Just try your best to fill in the most accurate information.

**Review by kevinzwang — APPROVED** [review]
LGTM

**Review by maxmwang — APPROVED** [review]

**Pull request merged**